### ISSUE TITLE
1.7.4 version can not be used in environment variables "."

### ISSUE THREAD

**Issue by liuzhen9327** [kubernetes/kubernetes#52381]
### Environment:
```
 - name: eureka.instance.hostname
    value: "service-registry"
 - name: eureka.server.enable-self-preservation
    value: "false"
 - name: eureka.client.serviceUrl.defaultZone
    value: "http://service-registry:8761/eureka/"
```
### Error message：
```
[root@master yaml]# kubectl create -f service-registry.yaml
service "service-registry" created
The Deployment "service-registry" is invalid:
* spec.template.spec.containers[0].env[0].name: Invalid value: "eureka.instance.hostname": a valid C identifier must start with alphabetic character or '_', followed by a string of alphanumeric characters or '_' (e.g. 'my_name',  or 'MY_NAME',  or 'MyName', regex used for validation is '[A-Za-z_][A-Za-z0-9_]*')
* spec.template.spec.containers[0].env[1].name: Invalid value: "eureka.server.enable-self-preservation": a valid C identifier must start with alphabetic character or '_', followed by a string of alphanumeric characters or '_' (e.g. 'my_name',  or 'MY_NAME',  or 'MyName', regex used for validation is '[A-Za-z_][A-Za-z0-9_]*')
* spec.template.spec.containers[0].env[2].name: Invalid value: "eureka.client.serviceUrl.defaultZone": a valid C identifier must start with alphabetic character or '_', followed by a string of alphanumeric characters or '_' (e.g. 'my_name',  or 'MY_NAME',  or 'MyName', regex used for validation is '[A-Za-z_][A-Za-z0-9_]*')
[root@master yaml]# kubectl get node
NAME      STATUS    AGE       VERSION
master    Ready     1d        v1.7.4+coreos.0
node1     Ready     1d        v1.7.4+coreos.0
node2     Ready     1d        v1.7.4+coreos.0
[root@master yaml]# kubectl --version
Kubernetes v1.7.4+coreos.0
```
[is not it fixed here?](https://github.com/kubernetes/kubernetes/pull/48986)

**Comment by liuzhen9327** [comment]
@k8s-merge-robot sorry，I won't add label.. can you help me?

**Comment by dixudx** [comment]
/sig apps

> is not it fixed here?

@liuzhen9327 This fix is not included in `v1.7.4`, nor `v1.7.5` . Please use `master` branch for now.

**Comment by liuzhen9327** [comment]
@dixudx thx! will the next version be included?

**Comment by dixudx** [comment]
Yes, will be included in release `v1.8`. You can have a try in current released `v1.8.0-beta.1`.

**Comment by liuzhen9327** [comment]
thx

**Comment by timoreimann** [comment]
As mentioned, the feature shipped in 1.8.

Please close this issue. Thanks.
